src/lib.rs

```diff
@@ -3589,10 +3589,10 @@ impl Build {
                     "x86_64-unknown-netbsd" => Some("x86_64--netbsd"),
                     "xtensa-esp32-espidf" => Some("xtensa-esp32-elf"),
                     "xtensa-esp32-none-elf" => Some("xtensa-esp32-elf"),
-                    "xtensa-esp32s2-espidf" => Some("xtensa-esp32s2-elf"),
-                    "xtensa-esp32s2-none-elf" => Some("xtensa-esp32s2-elf"),
-                    "xtensa-esp32s3-espidf" => Some("xtensa-esp32s3-elf"),
-                    "xtensa-esp32s3-none-elf" => Some("xtensa-esp32s3-elf"),
+                    "xtensa-esp32s2-espidf" => Some("xtensa-esp32-elf"),
```

Comment:
sorry if that is a bit confusing but its not xtensa-esp32-elf but just xtensa-esp-elf :D and all the wrappers for esp32 (legacy), esp32s2 and esp32s3 points to the xtensa-esp-elf variant. Hope that makes sense